lib/src/Base/Graph/GraphImplementation.cxx

```diff
@@ -144,7 +144,11 @@ String GraphImplementation::__repr__() const
 /* Adds a drawable instance to the collection of drawables contained in GraphImplementation */
 void GraphImplementation::add(const Drawable & aDrawable)
 {
+  UnsignedInteger drawableCount = drawablesCollection_.getSize();
   drawablesCollection_.add(aDrawable);
+  Drawable&drawable = drawablesCollection_[drawableCount];
+  if (!drawable.isColorExplicitlySet())
+    drawable.setColor(DrawableImplementation::BuildDefaultPalette(drawableCount + 1)[drawableCount]);
```

Comment:
its probably not very efficient to build the whole palette just to the last one
I just ask we add a TODO comment stating that we could add a method to retrieve the i-th color independently